### PR TITLE
Add csharp package

### PR DESCRIPTION
**Pull request by andrewpollock**
This is for #3087

### CODE REVIEW

**Comment by jtattermusch** [comment]
Is there a way to test the resulting package?

**Comment by jtattermusch** [comment]
I am not sure about the -cil package suffix, I feel it will be confusing to the users if the name doesn't mention "csharp" or "csharp extension" explicitly. What exactly is the rationale behind -cil suffix?

File: debian/control [hunk 1]
```diff
@@ -29,3 +29,14 @@ Description: high performance general RPC framework
  transparently, and makes it easier to build connected systems.
  .
  This package provides the gRPC C bindings.
+
+Package: libgrpc0-cil
+Architecture: any
+Section: cli-mono
+Depends: ${shlibs:Depends}, ${misc:Depends}
```
Comment (jtattermusch) [inline debian/control]: I think the libgrpc_csharp_ext.so depends on libgrpc and libgpr but dependencies don't state that.

File: debian/control [hunk 2]
```diff
@@ -29,3 +29,14 @@ Description: high performance general RPC framework
  transparently, and makes it easier to build connected systems.
  .
  This package provides the gRPC C bindings.
+
+Package: libgrpc0-cil
```
Comment (jtattermusch) [inline debian/control]: cil vs cli?

**Comment by andrewpollock** [comment]
> What exactly is the rationale behind -cil suffix?

I asked around (Debian) for what the naming convention should be for such packages, and was told that -cil (Common Intermediate Language) https://en.wikipedia.org/wiki/Common_Intermediate_Language was the convention.

File: debian/control (hunk 1 above)
Comment (andrewpollock) [inline debian/control]: This is the unexpanded control file. When I built the package, it expanded out to:
~~~
Depends: libc6 (>= 2.14), libgoogle-perftools4, libgrpc0, zlib1g (>= 1:1.1.4)
~~~

File: debian/control (hunk 2 above)
Comment (andrewpollock) [inline debian/control]: No, spelling is correct and consistent with other such packages. Check out
~~~
apt-cache search lib\.\*-cil
~~~

**Comment by andrewpollock** [comment]
> Is there a way to test the resulting package?

You could follow the instructions at https://github.com/andrewpollock/grpc/wiki/Debian-Package-Build to build the package locally and then you'd have a copy of the package.

File: debian/control (hunk 1 above)
Comment (jtattermusch) [inline debian/control]: Ack.

**Comment by jtattermusch** [comment]
Regarding the "-cil" suffix  - from looking at existing packages, it seems like most of the -cil packages contain .NET assemblies (.dll files) and only some of the also contain support .so native libraries. In our case, the package only contains the .so native library - the .NET assemblies for gRPC C# are distributed via a nuget package. Technically, our debian package doesn't contain any CLI binary and so name might be a bit misleading. On the other hand, it is clearly a package that provides supports to be able to run gRPC with CLI.
I think keeping the name you suggested would be alright, I just wanted to let you know about the concerns I had.

**Comment by andrewpollock** [comment]
> I think keeping the name you suggested would be alright, I just wanted to let you know about the concerns I had.

Thanks. You know more about this than I do. There is one final level of sanity checking: because we're adding a new binary package, this will have to go through the NEW queue again. The FTP masters will have the most experience with package names and contents, so if they think it's inappropriate they'll reject the package.

File: debian/control [hunk 3]
```diff
@@ -29,3 +29,14 @@ Description: high performance general RPC framework
  transparently, and makes it easier to build connected systems.
  .
  This package provides the gRPC C bindings.
```
Comment (jtattermusch) [inline debian/control]: provided the gRPC C# bindings?

Comment (andrewpollock) [inline debian/control]: > provides the gRPC C# bindings?

No.

~~~
$ dpkg --contents /var/cache/pbuilder/result/libgrpc0_0.10.2-2_amd64.deb 
drwxr-xr-x root/root         0 2015-08-28 15:16 ./
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/lib/
-rw-r--r-- root/root    412536 2015-08-28 15:16 ./usr/lib/libgrpc.so.0.10.1.0
-rw-r--r-- root/root     50912 2015-08-28 15:16 ./usr/lib/libgpr.so.0.10.1.0
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/share/
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/share/doc/
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/share/doc/libgrpc0/
-rw-r--r-- root/root       301 2015-08-28 15:12 ./usr/share/doc/libgrpc0/changelog.Debian.gz
-rw-r--r-- root/root      2556 2015-08-28 15:12 ./usr/share/doc/libgrpc0/copyright
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/share/lintian/
drwxr-xr-x root/root         0 2015-08-28 15:16 ./usr/share/lintian/overrides/
-rw-r--r-- root/root        38 2015-08-28 15:12 ./usr/share/lintian/overrides/libgrpc0
lrwxrwxrwx root/root         0 2015-08-28 15:16 ./usr/lib/libgrpc.so.0 -> libgrpc.so.0.10.1.0
lrwxrwxrwx root/root         0 2015-08-28 15:16 ./usr/lib/libgpr.so.0 -> libgpr.so.0.10.1.0
~~~

**Comment by jtattermusch** [comment]
Thanks, merging.

**Pull request merged**